apps/aesophia/src/aeso_ast_infer_types.erl

```diff
@@ -341,7 +349,10 @@ infer_letrec(Env, {letrec, Attrs, Defs}) ->
             Expect = typesig_to_fun_t(TypeSig),
             unify(Got, Expect, {check_typesig, Name, Got, Expect}),
             solve_field_constraints(),
-            io:format("Checked ~s : ~s\n", [Name, pp(dereference_deep(Got))]),
+            PP = fun () -> io:format("Checked ~s : ~s\n",
+                                     [Name, pp(dereference_deep(Got))])
+                 end,
+            when_option(pp_types, PP),
```

Comment:
A bit awkward. Maybe `-define(DEBUG(Fmt, Args), when_option(pp_types, fun() -> io:format(Fmt, Args) end))).` or some such to make it nicer?